torch/csrc/StorageMethods.cpp

```diff
@@ -149,7 +150,15 @@ static PyObject* THPStorage_resize_(PyObject* self, PyObject* number_arg) {
 #endif
   } else if (device_type == at::kMeta) {
     at::native::resize_bytes_meta(storage.unsafeGetStorageImpl(), newsize);
-  } else if (device_type == at::kXPU || device_type == at::kPrivateUse1) {
+  } else if (device_type == at::kPrivateUse1) {
+    if (at::isPrivateUse1HooksRegistered()) {
+        at::GetPrivateUse1HooksInterface()->resizePrivateUse1Bytes(storage, newsize);
+    } else {
+        TORCH_CHECK(
```

Comment:
You can move this in to the default implementation of resizePrivateUse1Bytes so that it behaves the same as all other functions there.
Then you can remove the isPrivateUse1HooksRegistered() check altogether.

Reply:
> You can move this in to the default implementation of resizePrivateUse1Bytes so that it behaves the same as all other functions there. Then you can remove the isPrivateUse1HooksRegistered() check altogether.

I have considered this, but it seems that ``isPrivateUse1HooksRegistered`` is redundant? Looks like we can always check in the default implementation.